src/core/meshcop/tcat_agent.cpp

```diff
@@ -540,6 +547,97 @@ Error TcatAgent::HandlePing(const Message &aIncomingMessage,
     return error;
 }
 
+Error TcatAgent::HandleGetNetworkName(Message &aOutgoingMessage)
+{
+    Error       error  = kErrorNone;
+    const char *name   = otThreadGetNetworkName(&GetInstance());
+    uint16_t    length = static_cast<uint16_t>(strlen(name));
+    ot::Tlv     tlv;
+
+    VerifyOrExit(Get<ActiveDatasetManager>().IsCommissioned(), error = kErrorInvalidState);
+    VerifyOrExit(length > 0, error = kErrorInvalidState);
+
+    tlv.SetType(kTlvResponseWithPayload);
+    tlv.SetLength(static_cast<uint8_t>(length));
+    SuccessOrExit(error = aOutgoingMessage.Append(tlv));
+    SuccessOrExit(error = aOutgoingMessage.AppendBytes(name, length));
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetDeviceId(Message &aOutgoingMessage)
```

Comment:
My suggestion to simplify the `HandleGetDeviceId`()

```C++
Error TcatAgent::HandleGetDeviceId(Message &aOutgoingMessage)
{
    uint16_t        length = 0;
    const uint8_t  *deviceId;
    Mac::ExtAddress eui64;

    static_assert(sizeof(Mac::ExtAddress) > OT_TCAT_MAX_VENDORID_SIZE, "TCAT VendorID size is greater than ExtAddress");

    if ((mVendorInfo->mDeviceIds != nullptr) && (mVendorInfo->mDeviceIds[0].mDeviceIdType != OT_TCAT_DEVICE_ID_EMPTY))
    {
        length   = static_cast<uint8_t>(mVendorInfo->mDeviceIds[0].mDeviceIdLen);
        deviceId = mVendorInfo->mDeviceIds[0].mDeviceId;
    }

    if (length == 0)
    {
        // If not found, or empty device ID, use EUI64.
        Get<Radio>().GetIeeeEui64(eui64);

        length   = sizeof(Mac::ExtAddress);
        deviceId = eui64.m8;
    }

    return Tlv::AppendTlv(aOutgoingMessage, kTlvResponseWithPayload, deviceId, length);
}
```

- Can track the `deviceId` as `const uint8_t *` (no need to copy)
- Remove `for` loop
- Use `Get<Radio>().GetIeeeEui64(deviceId)`
- Use `Tlv::AppendTlv()`

Reply:
Another idea:

We can also consider removing the `static_assert(sizeof(Mac::ExtAddress) > OT_TCAT_MAX_VENDORID_SIZE)` check, as we're no longer copying the `mDeviceId` bytes into a separate array and are directly reading it from `mVendorInfo->mDeviceIds[0]`.  We only need to make sure it fits in a TLV (less than `Tlv::kBaseTlvMaxLength`).

Reply:
@EskoDijk  @abtink  I have introduced the distinction between advertised device IDs and returned by query over BLE. Since those are clearly two different things.

This change also includes refactoring of HandleGetDeviceId similar to advised by @abtink . 
This is why I resolved rest of the comments related to this part of the code.